### ISSUE TITLE
SpStringToColor fails for HTML strings

### ISSUE THREAD

**Issue by pyscripter** [SilverpointDev/sptbxlib#3]
This is currently the code in TSpTBXSkins.pas
```
function SpStringToColor(S: string; out Color: TColor): Boolean;
var
  L, C: Integer;
begin
  Result := False;
  Color := clDefault;
  L := Length(S);
  if L < 2 then Exit;

  if (S[1] = '#') and (L = 7) then begin
    Delete(S, 1, 1); // strip the # char
    if TryStrToInt(S, C) then begin
      S := Format('$00%s%s%s', [Copy(S, 5, 2), Copy(S, 3, 2), Copy(S, 1, 2)]);
      Color := StringToColor(S);
      Result := True;
    end;
  end
```
As you see in the code if S[1] ='#' you delete it and try to convert the rest to Int.  The conversion will fail if the HTML color constant contains letters.
To solve the issue replace the above with the following:
```

function SpStringToColor(S: string; out Color: TColor): Boolean;
var
  L, C: Integer;
begin
  Result := False;
  Color := clDefault;
  L := Length(S);
  if L < 2 then Exit;

  if (S[1] = '#') and (L = 7) then begin
    S[1] := '$';
    if TryStrToInt(S, C) then begin
      S := Format('$00%s%s%s', [Copy(S, 6, 2), Copy(S, 4, 2), Copy(S, 2, 2)]);
      Color := StringToColor(S);
      Result := True;
    end;
  end
```

**Comment by madorin** [comment]
Good point.
Also, for ex #BAD is a valid color too :) (shorthand hexadecimal form)

**Comment by SilverpointDev** [comment]
Fixed in 2.5.4
